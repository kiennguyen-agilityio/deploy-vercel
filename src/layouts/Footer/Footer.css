.footer-distributed {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  background: var(--color-white);
  padding-top: 114px;
}

.logo-footer {
  display: flex;
  align-items: center;
  margin-bottom: 80px;
}

.logo i {
  font-size: var(--font-size-md);
  margin-right: 5px;
}

.shop-name {
  font-weight: bold;
}

.social-icons {
  margin: 0 0 80px 30px;
}

.social-icons a {
  margin-right: 30px;
}

.social-icons i {
  font-size: var(fs-large-more);
  color: var(--color-black);
}

.copyright {
  margin-top: 10px;
  text-align: center;
}
