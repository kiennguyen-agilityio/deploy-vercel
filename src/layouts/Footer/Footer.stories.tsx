import { StoryObj, Meta } from '@storybook/react';
import { MemoryRouter } from 'react-router-dom';
import Footer from './Footer';

export default {
  title: 'Components/Footer',
  component: Footer,
  decorators: [
    (Story) => (
      <MemoryRouter>
        <Story />
      </MemoryRouter>
    ),
  ],
} as Meta;

type Story = StoryObj<typeof Footer>;

export const Default: Story = {};
