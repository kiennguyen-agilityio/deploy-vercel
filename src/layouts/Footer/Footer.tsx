import { memo } from 'react';

import './Footer.css';
import { Link } from 'react-router-dom';

const Footer = () => {
  return (
    <footer className="footer-distributed">
      <Link to="/">
        <div className="logo-footer">
          <i className="fa-sharp fa-solid fa-bag-shopping" />
          <span className="shop-name">Shop Bags</span>
        </div>
      </Link>

      <div className="social-icons">
        <a href="https://www.facebook.com">
          <i className="fa-brands fa-square-facebook" />
        </a>
        <a href="https://www.twitter.com">
          <i className="fa-brands fa-square-twitter" />
        </a>
        <a href="https://www.linkedin.com">
          <i className="fa-brands fa-linkedin" />
        </a>
        <a href="https://www.telegram.org">
          <i className="fa-brands fa-square-instagram" />
        </a>
      </div>
      <p className="copyright">2019 Â© Shopbag - All right reserved</p>
    </footer>
  );
};

export default memo(Footer);
