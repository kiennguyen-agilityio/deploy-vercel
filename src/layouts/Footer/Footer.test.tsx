import { render } from '@testing-library/react';
import Footer from './Footer';

describe('Footer component', () => {
  it('renders correctly', () => {
    const { container } = render(<Footer />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('display shop name', () => {
    const { getByText } = render(<Footer />);
    const shopName = getByText('Shop Bags');
    expect(shopName).toBeInTheDocument();
  });

  it('displays copyright notice', () => {
    const { getByText } = render(<Footer />);
    const copyright = getByText('2019 Â© Shopbag - All right reserved');
    expect(copyright).toBeInTheDocument();
  });
});
