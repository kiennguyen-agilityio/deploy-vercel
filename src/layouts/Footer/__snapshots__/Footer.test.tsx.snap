// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component renders correctly 1`] = `
<footer
  class="footer-distributed"
>
  <div
    class="logo-footer"
  >
    <i
      class="fa-sharp fa-solid fa-bag-shopping"
    />
    <span
      class="shop-name"
    >
      Shop Bags
    </span>
  </div>
  <div
    class="social-icons"
  >
    <a
      href="https://www.facebook.com"
    >
      <i
        class="fa-brands fa-square-facebook"
      />
    </a>
    <a
      href="https://www.twitter.com"
    >
      <i
        class="fa-brands fa-square-twitter"
      />
    </a>
    <a
      href="https://www.linkedin.com"
    >
      <i
        class="fa-brands fa-linkedin"
      />
    </a>
    <a
      href="https://www.telegram.org"
    >
      <i
        class="fa-brands fa-square-instagram"
      />
    </a>
  </div>
  <p
    class="copyright"
  >
    2019 © Shopbag - All right reserved
  </p>
</footer>
`;

exports[`renders correctly 1`] = `
<footer
  class="footer-distributed"
>
  <div
    class="logo-footer"
  >
    <i
      class="fa-sharp fa-solid fa-bag-shopping"
    />
    <span
      class="shop-name"
    >
      Shop Bags
    </span>
  </div>
  <div
    class="social-icons"
  >
    <a
      href="https://www.facebook.com"
    >
      <i
        class="fa-brands fa-square-facebook"
      />
    </a>
    <a
      href="https://www.twitter.com"
    >
      <i
        class="fa-brands fa-square-twitter"
      />
    </a>
    <a
      href="https://www.linkedin.com"
    >
      <i
        class="fa-brands fa-linkedin"
      />
    </a>
    <a
      href="https://www.telegram.org"
    >
      <i
        class="fa-brands fa-square-instagram"
      />
    </a>
  </div>
  <p
    class="copyright"
  >
    2019 © Shopbag - All right reserved
  </p>
</footer>
`;
